packages/web3-eth/types/index.d.ts

```diff
@@ -0,0 +1,268 @@
+    signTransaction(transaction: Transaction, address: string): Promise<SignedTransaction>;
+    signTransaction(transaction: Transaction, address: string, callback: (error: Error, signedTransaction: SignedTransaction) => void): Promise<SignedTransaction>;
+    call(transaction: Transaction): Promise<string>;
+    call(transaction: Transaction, defaultBlock?: number | string): Promise<string>;
+    call(transaction: Transaction, callback?: (error: Error, data: string) => void): Promise<string>;
+    call(transaction: Transaction, defaultBlock: number | string, callback: (error: Error, data: string) => void): Promise<string>;
+    estimateGas(transaction: Transaction, callback?: (error: Error, gas: number) => void): Promise<number>;
+    getPastLogs(options: PastLogsOptions, callback?: (error: Error, logs: Log[]) => void): Promise<Log[]>;
+    getWork(callback?: (error: Error, result: string[]) => void): Promise<string[]>;
+    submitWork(data: [string, string, string], callback?: (error: Error, result: boolean) => void): Promise<boolean>;
+}
+
+export interface Methods {
+    property?: string;
+    methods: Method[];
+}
+
+export interface Method {
+    name: string;
+    call: string;
+    params?: number;
+    inputFormatter?: Array<(() => void) | null>;
+    outputFormatter?: () => void;
+}
+
+export interface Syncing {
+    startingBlock: number;
+    currentBlock: number;
+    highestBlock: number;
+    knownStates: number;
+    pulledStates: number;
+}
+
+export interface BlockHeader {
+    number: number
+    hash: string
+    parentHash: string
+    nonce: string
+    sha3Uncles: string
+    logsBloom: string
+    transactionRoot: string
+    stateRoot: string
+    receiptRoot: string
+    miner: string
+    extraData: string
+    gasLimit: number
+    gasUsed: number
+    timestamp: number
+}
+
+export interface Block extends BlockHeader {
+    transactions: Transaction[];
+    size: number
+    difficulty: number
+    totalDifficulty: number
+    uncles: string[];
+}
+
+export interface SignedTransaction {
+    raw: string;
+    tx: Transaction;
+}
+
+export interface PastLogsOptions {
+    fromBlock?: number | string;
+    toBlock?: number | string;
+    address: string | string[];
+    topics?: Array<string | string[]>;
+}
+
+export interface Logs {
+    fromBlock?: number
+    address?: string
+    topics?: Array<string | string[]>
+}
+
+export interface Log {
+    address: string;
+    data: string;
+    topics: Array<string | string[]>;
+    logIndex: number;
+    transactionIndex: number;
+    transactionHash: string;
+    blockHash: string;
+    blockNumber: number;
+}
+
+export interface Subscribe<T> {
+    subscription: {
+        id: string
+        subscribe(callback?: (error: Error, result: Subscribe<T>) => void): Subscribe<T>
+        unsubscribe(callback?: (error: Error, result: boolean) => void): void | boolean
+        arguments: object
```

Comment:
It's called options and not arguments :)
You could create an AbstractSubscription type: 
https://github.com/ethereum/web3.js/blob/ethereumProvider/packages/web3-core-subscriptions/lib/subscriptions/AbstractSubscription.js